compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -756,7 +758,10 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                                 trait_ref,
                                 span,
                             ) {
-                                GetSafeTransmuteErrorAndReason::Silent => return,
+                                GetSafeTransmuteErrorAndReason::ErrorGuaranteed(_guar) => return,
+                                // Unable to compute whether Safe Transmute is possible (for example, due to an unevaluated const).
+                                // The same thing occurred during trait selection/confirmation, so there is no error to report here.
```

Comment:
this feels somewhat dangerous to me :shrug: i guess it's fine because fulfillment error reporting uses `delay_span_bug` so if it's wrong we just ICE